### PR TITLE
names in the config will override default lock name

### PR DESCRIPTION
**Pull request by julien51**
# Description

This was reported by an early of the new paywall :)
We now make sure to overide the name of the lock with the config.

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] I have updated the docs to reflect my changes if applicable
- [X] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by julien51** [comment]
Hum. I think there might a normalization here... annoying :/

File: unlock-app/src/hooks/usePaywallLocks.ts [hunk 1]
```diff
@@ -31,6 +32,14 @@ export const usePaywallLocks = (
 
       // getLock doesn't always return the lock address apparently
       lock.address = lockAddresses[index]
+
+      // Use the config's overwrite for the name if applicable
+      Object.keys(config.locks).forEach(configLockAddress => {
+        // The address may differ from the config to the lock received from web3Service.
+        if (configLockAddress.toLowerCase() === lock.address.toLowerCase()) {
```
Comment (julien51) [inline unlock-app/src/hooks/usePaywallLocks.ts:39]: I think this solves the normalization issue.

Comment (julien51) [reply unlock-app/src/hooks/usePaywallLocks.ts:39]: weird. It looks like it does not :/

Comment (cnasc) [reply unlock-app/src/hooks/usePaywallLocks.ts:39]: We have a `foreach` over the locks inside a `foreach` over the locks here. I think we can simplify this by just using the lock address already available inside each iteration of the existing loop to index into the config and pull the override name.

Comment (julien51) [reply unlock-app/src/hooks/usePaywallLocks.ts]: Good point!

**Pull request merged**